egui/src/widgets/plot/mod.rs

```diff
@@ -20,12 +20,48 @@ mod items;
 mod legend;
 mod transform;
 
-type CustomLabelFunc = dyn Fn(&str, &Value) -> String;
-type CustomLabelFuncRef = Option<Box<CustomLabelFunc>>;
-
-type AxisFormatterFn = dyn Fn(f64) -> String;
+type LabelFormatterFn = dyn Fn(&str, &Value) -> String;
+type LabelFormatter = Option<Box<LabelFormatterFn>>;
+type AxisFormatterFn = dyn Fn(f64, &RangeInclusive<f64>) -> String;
 type AxisFormatter = Option<Box<AxisFormatterFn>>;
 
+/// Specifies the coordinates formatting when passed to [`Plot::coordinates_formatter`].
+pub struct CoordinatesFormatter {
+    function: Box<dyn Fn(&Value, &PlotBounds) -> String>,
+}
+
+impl CoordinatesFormatter {
+    /// Create a new formatter based on the pointer coordinate and the plot bounds.
+    pub fn new(function: impl Fn(&Value, &PlotBounds) -> String + 'static) -> Self {
+        Self {
+            function: Box::new(function),
+        }
+    }
+
+    /// Show a fixed number of decimal places.
+    pub fn with_precision(precision: usize) -> Self {
```

Comment:
"precision" and "decimal places" are not the same thing. `42.3` has a precision=3, but only one decimal place.

```suggestion
    /// Show a fixed number of decimal places.
    pub fn with_decimals(num_decimals: usize) -> Self {
```